pdfgen/src/types/hierarchy/trailer.rs

```diff
@@ -32,9 +32,9 @@ impl WriteTrailer for CrossReferenceTable {
         root: ObjId,
         id: [u8; 16],
     ) -> Result<(), std::io::Error> {
-        const SIZE: Name = Name::new(b"Size");
-        const ROOT: Name = Name::new(b"Root");
-        const ID: Name = Name::new(b"ID");
+        const SIZE: Name<'static> = Name::from_static(b"Size");
+        const ROOT: Name<'static> = Name::from_static(b"Root");
+        const ID: Name<'static> = Name::from_static(b"ID");
```

Comment:
In general we have a lot of repetition in these declarations. I would propose that we try to write a procedural function-like macro (in a separate crate `pdfgen_macros`). We could then have something like this: 

```rust
// this would expand to the same code. 
const_names!(SIZE, ROOT, ID);
```

This would make the generalization of `Name` work great, as we wouldn't have to write (even more) verbose `const XXX: Name<&'static [u8]> = Name::from_static("Xxx");`